airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -695,10 +699,12 @@ def parent_hash_id(self) -> str:
 
     def unnesting_before_query(self) -> str:
         if self.parent and self.is_nested_array:
-            parent_table_name = f"'{self.parent.actual_table_name()}'"
+            parent_file_name = (
+                f"'{self.get_file_name(self.parent.stream_name, self.parent.get_schema(False), self.parent.actual_table_name())}'"
```

Comment:
Chris, can you give a quick summary what is happening here? One thing I did not understand when I was reading this code was the unnesting bits.

Reply:
In this function, we format a string to do a call to jinja macro in the generated dbt model at line 707:

> jinja_call(f"unnest_cte({parent_file_name}, {parent_stream_name}, {quoted_field})")

where the first argument is expected to be a "table_name" as used in `ref('<table_name>')` calls inside the macro: 
https://github.com/airbytehq/airbyte/blob/b0157f04d1996bc031a152b58103ef604e0b7f2a/airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/array.sql#L93

So since the convention of filenames changed, this needed to be propagated through the `parent_file_name` too

Otherwise, runtime execution of this macro (only in unnesting on redshift) would fail

Reply:
I struggled to find the macro since Intellij doesn't detect it in autocomplete. this is helpful, thanks.